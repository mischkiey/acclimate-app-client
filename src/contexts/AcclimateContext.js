import React from 'react';

const AcclimateContext = React.createContext({});

export default AcclimateContext;