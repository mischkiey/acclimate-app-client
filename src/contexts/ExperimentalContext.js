import React, { Component } from 'react';
import TokenService from '../services/token-services';

export const ExperimentalContext = React.createContext({});

export class ExperimentalProvider extends Component {
    state = {
        log: false,
        tasks: [],
        shoppingItems: [],
    };

    handleUserLog = () => {
        const log = TokenService.hasAuthToken();
        this.setState({log});
    };

    handleUserLogOut = () => {
        TokenService.clearAuthToken();
        this.handleUserLog();
    };

    setTasks = (tasks) => {
        this.setState({tasks})
    };

    setShoppingItems = (shoppingItems) => {
        this.setState({shoppingItems})
    };

    render() {
        const value = {
            log: this.state.log,
            tasks: this.state.tasks,
            shoppingItems: this.state.shoppingItems,

            handleUserLog: this.handleUserLog,
            handleUserLogOut: this.handleUserLogOut,
            setTasks: this.setTasks,
            setShoppingItems: this.setShoppingItems,
        };

        return (
            <ExperimentalContext.Provider value={value}>
                {this.props.children}
            </ExperimentalContext.Provider>
        );
    };
};

