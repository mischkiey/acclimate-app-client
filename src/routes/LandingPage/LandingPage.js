import React from 'react';
import './LandingPage.css';

function LandingPage () {
    return (
        <>
            <h2>Description:</h2>
            <p>Acclimate is an informational and utility disaster management app that aims to provide accurate, reliable, and useful information and tools to be able to handle impending disasters systematically and effectively</p>
            
            <h2>How It Works:</h2>
            <ol>
                <li>The app will provide users with general information on a selected disaster</li>
                <li>The app will generate a disaster management plan (based on user-selection) that has steps for before, during, and after the event</li>
                <li>The app will also generate a task list that incorporates the above plan</li>
                <li> The app will also generate a shopping list with recommended supplies</li>
                <li> Users can personalize the task and shopping lists to tailorfit their current circumstances/needs</li>
            </ol>
        </>
    );
};

export default LandingPage;