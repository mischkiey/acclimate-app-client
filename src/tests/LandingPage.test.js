import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { AcclimateProvider } from '../contexts/AcclimateContext';
import LandingPage from '../routes/LandingPage/LandingPage';

describe(`LandingPage Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <AcclimateProvider>
                        <LandingPage />
                    </AcclimateProvider>
                </BrowserRouter>,
                div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <AcclimateProvider>
                            <LandingPage />
                        </AcclimateProvider>
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})