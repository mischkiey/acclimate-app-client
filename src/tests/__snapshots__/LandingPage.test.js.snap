// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="white"
>
  <i
    className="material-icons"
  >
    face
  </i>
  <h2>
    Hello and welcome to Acclimate!
  </h2>
  <p>
    Acclimate is an informational and utility disaster management app that aims to provide accurate, reliable, and useful information and tools to be able to handle impending disasters systematically and effectively
  </p>
  <hr />
  <h2>
    How It Works:
  </h2>
  <ol>
    <li>
      The app provides the user with general information on a selected disaster
    </li>
    <li>
      The app provides disaster management plans that has steps for before, during, and after the event
    </li>
    <li>
      The app provides the user with a highly-customizable task list and shopping list
    </li>
    <li>
      Users may add steps from the chosen disaster plan to their lists where appropriate at the push of a button
    </li>
  </ol>
  <hr />
  <p>
    Click 
    <a
      href="/signuppage"
      onClick={[Function]}
    >
      'Signup'
    </a>
     to create an account and get started!
  </p>
</div>
`;
